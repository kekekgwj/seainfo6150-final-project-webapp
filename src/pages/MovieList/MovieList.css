.movie-list-wrapper {
    margin: 0 auto;
}
.movie-list-now {
    margin-top: 80px;
}
.movie-list-content {
    display: flex;
    flex-direction: column;
    align-items: center;
}

@media only screen and (min-width: 0px) {
    /* For tablets: */
    .movie-list-wrapper {
        width: 100%;
    }

}
@media only screen and (min-width: 500px) {
    /* For tablets: */
    .movie-list-wrapper {
        width: 500px;
    }
    .movie-list-popular {
        padding-bottom: 400px;
    }

}
@media only screen and (min-width: 992px) {
    /* For tablets: */
    .movie-list-wrapper {
        width: 800px;
    }
    .movie-list-popular {
        padding-bottom: 0px;
    }

}
  
@media only screen and (min-width: 1200px) {
    /* For desktop: */
    .movie-list-wrapper {
        width: 1000px;
    }
    .movie-list-popular {
        padding-bottom: 0px;
    }
}
