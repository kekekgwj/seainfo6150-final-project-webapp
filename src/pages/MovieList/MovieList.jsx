import React, { useEffect, useState } from "react";
import { getMovieListData } from "../../services";
import "./MovieList.css";
import MovieListBox from  "../../components/MovieListBox/MovieListBox"


const MovieList = () => {
    const [data, setData] = useState({});
    
    useEffect(() => {
        getMovieListData()
        .then( res => {
            setData(res);
        })
        .catch(err => {
            console.log(err);
        })

    }, [])

    return (
        <div className="movie-list-wrapper">
            <div className="movie-list-content">
                <div className="movie-list-popular"> 
                    <MovieListBox movieListData={data.now} title="Now Showing" />
                </div>
                <div className="movie-list-now"> 
                    <MovieListBox movieListData={data.pop} title="Popular Movie"/>
                </div>
                
            </div>
        </div>
    )
}

export default MovieList;