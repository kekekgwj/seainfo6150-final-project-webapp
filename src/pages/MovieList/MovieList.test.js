import React from "react";
import MovieList from "./MovieList";

describe("Movie List page tests", () => {
  it("renders correctly", () => {
    const { container } = render(<MovieList/>);
    expect(container).toMatchSnapshot();
  });
});