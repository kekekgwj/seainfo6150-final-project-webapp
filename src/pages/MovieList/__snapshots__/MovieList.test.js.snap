// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie List page tests renders correctly 1`] = `
<div>
  <div
    class="movie-list-wrapper"
  >
    <div
      class="movie-list-content"
    >
      <div
        class="movie-list-popular"
      >
        <div
          class="movie-list-box-container"
        >
          <p
            class="movie-list-box-title"
          >
            Now Showing
          </p>
          <div
            class="snippet-list-wrapper"
          />
        </div>
      </div>
      <div
        class="movie-list-now"
      >
        <div
          class="movie-list-box-container"
        >
          <p
            class="movie-list-box-title"
          >
            Popular Movie
          </p>
          <div
            class="snippet-list-wrapper"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
